File: wrappers/dotnet/src/Hyperledger.Ursa.BbsSignatures/Keys/BbsKey.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Hyperledger.Ursa.BbsSignatures
{
    /// <summary>
    /// A BBS+ key pair
    /// </summary>
    public class BbsKey
    {
        public BbsKey(byte[] publicKey, uint messageCount)
        {
            PublicKey = new ReadOnlyCollection<byte>(publicKey);
            MessageCount = messageCount;
        }

        /// <summary>
        /// Raw public key value for the key pair
        /// </summary>
        /// <returns></returns>
        public ReadOnlyCollection<byte> PublicKey { get; }

        /// <summary>
        /// Number of messages that can be signed
        /// </summary>
        /// <value>
        /// The message count.
        /// </value>
        public uint MessageCount { get; }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/keys/BbsKey.py
class BbsKey:
    def __init__(self, public_key: bytes, message_count: int) -> None:
        """ A BBS+ key pair

        Args:
            public_key: Raw public key value for the key pair
            message_count: Number of messages that can be signed
        """
        self.public_key = public_key
        self.message_count = message_count
